packages/vscode-extension/src/utilities/appCaches.ts

```diff
@@ -0,0 +1,97 @@
+import { join } from "path";
+import fs from "fs";
+import { getOrCreateAppsCachesDir } from "./common";
+import { createHash } from "crypto";
+import { getAppRootFolder } from "./extensionContext";
+import { Logger } from "../Logger";
+
+function getAppRootPathHash() {
+  return createHash("md5").update(getAppRootFolder()).digest("hex");
+}
+
+function getAppCachesPath() {
+  const appCachesPath = join(getOrCreateAppsCachesDir(), getAppRootPathHash());
+  return appCachesPath;
+}
+
+/**
+ * This function gets app level caches from caches storage.
+ * @param key
+ * @returns the value of the cache for key
+ */
+export function getAppCache(key: string) {
+  const appCachesPath = getAppCachesPath();
+  let caches;
+  try {
+    caches = JSON.parse(fs.readFileSync(appCachesPath).toString());
+  } catch (e) {
+    Logger.warn(`Error parsing caches file for app ${getAppRootFolder()}`, e);
+    return undefined;
+  }
+  if (!caches[key]) {
+    Logger.debug(`No app caches found for key: ${key}}`);
+    return undefined;
+  }
+  return caches[key];
+}
+
+/**
+ * This function sets app level caches from caches storage.
+ * @param key
+ * @param value
+ * @returns the new value that was set
+ */
+export function setAppCache(key: string, value: string) {
```

Comment:
this whole module doesn't have a clear contract. It seem to assume that the values it saves or retrieves are strings, but internal implementation parses those files expecting this is actually a structured data

Reply:
I stores strings for given key but the whole caches file is a json